### ISSUE TITLE
Doc2Vec inferring text of pretrained text not close enough

### ISSUE THREAD

**Issue by wobu** [deeplearning4j/deeplearning4j#3077]
Version: 0.8.0

When inferring a vector with the same text a label was pretrained, i expect the inferred vector is close to the vector of the pretrained label.
Currently the resulting similarity is always close to zero (indeed i had never a result better than this).

So i tried to compare results with a gensim example, where the closest labels of an inferred vector are always the pretrained labels.
[Gensim IMDB Example](https://github.com/RaRe-Technologies/gensim/blob/develop/docs/notebooks/doc2vec-IMDB.ipynb)

![image](https://cloud.githubusercontent.com/assets/649699/24106761/33c5ee3e-0d89-11e7-9508-69ea28d036b3.png)

I tried to imitate the Doc2Vec settings in Dl4J with the same dataset, but the inferred vector is always unclose to the pretrained vector (close to zero)

One setting which irritates me is the iteration parameter in dl4j, in all other implementations, are only existing epochs, which differs from the iterations in dl4j. But when using epochs in dl4j it seems that the learning rate isn't adapted correctly.

I created a [**Gist**](https://gist.github.com/wobu/8d996928e71afca7e4a1c117d7fc98a8) for setting up the testdata and training with dl4j.

**Comment by tezer** [comment]
Any progress so far?

**Comment by wobu** [comment]
tried with 

    .usePreciseWeightInit(true)

and other different configuratios (with DM, DBOW).

-> Also non similiar
